rasa/core/channels/channel.py

```diff
@@ -393,11 +393,21 @@ async def send_text_with_buttons(
         buttons: List[Dict[Text, Any]],
         **kwargs: Any,
     ) -> None:
-        await self._persist_message(
-            self._message(recipient_id, text=text, buttons=buttons)
-        )
+        """Sends text message with buttons."""
+        message_parts: List[Text] = text.strip().split("\n\n")
+        last_message_id: int = len(message_parts)
```

Comment:
This variable name is slightly misleading:
- it's not the ID of the last message, but the index of the last message
- However, it's also not really the index of the last message either, and only becomes that index because of the `start=1` below.

This is more of a nitpick, but I'd suggest to remove the `start=1`, changing the arithmetic accordingly, and renaming this variable.

Reply:
Fixed this. It looks better now, thank you!